developers/prerequisites/osgi.md

```diff
@@ -99,6 +99,21 @@ A bundle can therefore register a service, it can get a service and it can track
 ![OSGi Services][fig3]  
 Fig.3 OSGi Services (Source: <https://www.osgi.org/wp-content/uploads/services.png>)
 
+Popular OSGi Containers
+-----------------------
+
+We have talked so far about the OSGi specification. Here is the place to emphasize that the different containers might implement different parts of the OSGi specifications and might provide slightly different API (which could make the bundle not fully portable between different containers).
+
+We will list the most popular OSGi containers with a short description of their goals. We can divide them into:
+
+- open source:
+  - [Equinox](https://www.eclipse.org/equinox/) - this is the reference implementation of the OSGi R4.x Core Specification and one of the mostly used. As it is used in the openHAB project, we have prepared a  [wiki page](equinox.html); 
+  - [Apache Felix](http://felix.apache.org/) - implements OSGi R5 Core Specification, developed by the Apache Software Foundation. [Apache Karaf](http://karaf.apache.org/) is distribution based on Apache Felix that provides some additional features on top of it (e.g. folder based hot deployment, improved default console with remote SSH, maven plugins and others);
+  - [Concierge](https://www.eclipse.org/concierge/) - implements OSGi R5 Core Specification and is aimed at mobile and embedded devices. With a size of around 250 kb it is the smallest;

```

Comment:
"it is the smallest" is a bit incomplete for me. We can finish the sentence with something like "it has the smallest footprint of the presented containers".